src/zepben/evolve/model/busbranch/bus_branch.py

```diff
@@ -0,0 +1,343 @@
+#  Copyright 2020 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from collections import defaultdict
+from enum import Enum
+from functools import reduce
+from typing import Set, Tuple, FrozenSet, Dict, Callable, Union, TypeVar, Optional
+
+from zepben.evolve import NetworkService, ConnectivityNode, Terminal, Switch, AcLineSegment, ConductingEquipment, \
+    PowerTransformer, \
+    EnergySource, EnergyConsumer, IdentifiedObject, ShuntCompensator, Junction, BusbarSection, Jumper, \
+    PerLengthSequenceImpedance, WireInfo
+
+__all__ = ["create_bus_branch_model", "CreationResult", "ErrorInfo", "ErrorType"]
+
+BBN = TypeVar('BBN')  # Bus-Branch Network
```

Comment:
what exactly is BBN? is it a NetworkService or just a dictionary of stuff?

Reply:
So it is a generic type that represents a "bus branch network". Different tools will have different classes to represent a bus-branch-network for example pandapower has the "pp.pandapowerNet", PSSE will have a different one, etc.

The same thing happens for the different generic types, TN which represents a "Topological Node" is represented by a Bus class in pandapower, by a Node class in  Sincal, etc.

From the points of view of the code it doesn't really care what it is, it just calls factory methods that create instances of these things and knows where to passes theses instances around.

Reply:
okay sure